tomusborne/generateblocks
# Remove empty button container

Blocked by #234 

This PR checks the number of buttons inside the button container. If no buttons exist, it doesn't output the button container wrapper.

It also takes dynamic buttons into account. While the button container thinks it has buttons, it's possible that buttons have no dynamic data (no post meta, no terms). This PR removes those buttons from the count.

To test:

1. Add Query Loop
2. Add list of terms button
3. Add margin to button container

Without this PR you'll see an empty button container element (with margin taking up space) within posts with no terms.

With this PR no button container element should display if there are no buttons within it.
